src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+import argparse
+import grp
+import pwd
+import json
+import os
+import re
+import uuid
+import shutil
+import subprocess
+import platform
+import requests
+import zipfile
+from typing import Any, Dict, List, Literal, Optional, Sequence, Tuple
+from uuid import uuid1
+from boto3 import client
+from botocore.exceptions import ClientError
+import time
+import logging
+import yaml
+from subprocess import run
+from pathlib import Path
+from typing import Sequence, Optional, Any, Dict, List, Literal, Optional, Sequence, NamedTuple
+
+S3_ARTIFACT_DIR = "artifacts"
+DEVICE_PATH = "/var/lib/greengrass/device.pem.crt"
+PRIVATE_PATH = "/var/lib/greengrass/private.pem.key"
+CA_PATH = "/var/lib/greengrass/AmazonRootCA1.pem"
+JSON_FILE = "iot_setup_data.json"
+
+
+def install_greengrass_lite_from_source(commit_id: str, region: str):
+    # Get config
+    with open(JSON_FILE, 'r') as file:
+        data = json.load(file)
+
+    device_cert = data['DEVICE_CERT']
+    private_key = data['PRIVATE_KEY']
+    thing_name = data['THING_NAME']
+    thing_group_name = data['THING_GROUP_NAME']
+
+    # Download the source repo
+    download_result = _download_source(commit_id)
+    if not download_result:
+        return False
+
+    # Change the working dir
+    original_dir = os.getcwd()
+    os.chdir("aws-greengrass-lite")
+
+    # Set up an iot client
+    iot_client = client("iot", region_name=region)
+
+    try:
+
+        # Install build tools
+        install_result = _install_build_dependencies()
+        if not install_result:
+            return False
+
+        # Add user and group
+        add_result_ggcore = _add_user_and_group("ggcore", "ggcore")
+        add_result_gg_component = _add_user_and_group("gg_component",
+                                                      "gg_component")
+        if not add_result_ggcore or not add_result_gg_component:
+            return False
+
+        # Create several directories
+        dir1_result = _create_dir("/ggcredentials",
+                                  ownership=True,
+                                  flag='-R',
+                                  user="ggcore",
+                                  group="ggcore")
+        dir2_result = _create_dir("/var/lib/greengrass",
+                                  ownership=True,
+                                  user="ggcore",
+                                  group="ggcore")
+        dir3_result = _create_dir("/etc/greengrass")
+        if not dir1_result or not dir2_result or not dir3_result:
+            return False
+
+        # Build
+        build_result = _build_with_cmake()
+        if not build_result:
+            return False
+
+        # TES setup
+        tes_result = _tes_setup(device_cert, private_key)
+        if not tes_result:
+            return False
+
+        # Config setup
+        src_path = "docs/examples/sample_nucleus_config.yaml"
+        temp_path = "./config.yaml"
+        dest_path = "/etc/greengrass/config.yaml"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

 The absolute path starts at the root and moves down through each file. While traversing the path, if the folder name or location is different than the file path mentioned in the code, it causes a break. Since most of the computers have different directory structures, using absolute path in your code is likely to cause issues in your computational environment. We recommend using relative paths, because it only relies on relevant part of the directory structure being the same making it reliable to work with in computational environment